### PR TITLE
minor: create latest projects file for regression testing

### PR DESCRIPTION
**Pull request by nick-mancuso**
This is just a copy/paste/modify of the existing projects file, where specific commits and tags have been removed to allow us to test on the latest version of the project.

See https://github.com/checkstyle/checkstyle/pull/11805#issue-1291342040 and https://github.com/checkstyle/checkstyle/pull/11805#issuecomment-1172370264  for proof that Checkstyle can complete execution with these projects/ properties.

### CODE REVIEW

File: checkstyle-tester/latest-projects-to-test-on.properties [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# List of GIT repositories to clone / pull for checking with Checkstyle
+# File format: REPO_NAME|[local|git|hg]|URL|[COMMIT_ID]|[EXCLUDE FOLDERS]
+# Please note that bash comments works in this file
+
+guava|git|https://github.com/google/guava|v28.2||
```
Comment (rnveach) [inline checkstyle-tester/latest-projects-to-test-on.properties:5]: Why don't we just remove the specific versions from all of them and have them always be the latest?

Do we want to add this as a feature to the groovy script to ignore commit/tag?

Comment (nick-mancuso) [reply checkstyle-tester/latest-projects-to-test-on.properties]: > Guava still specifies a specific version.

Mistake, fixed.

> Do we want to add this as a feature to the groovy script to ignore commit/tag?

I would rather not do this, for a few reasons:
1. I would rather get by with current groovy script and put future effort into a whole new system
2. This still does not help us for check regression report in github CI, which is probably how 98% of reports are generated these days. 

We could add some options to github action that would translate to groovy script execution, but I would prefer to not complicate that more either.

Comment (nick-mancuso) [reply checkstyle-tester/latest-projects-to-test-on.properties]: Generating new reports at  https://github.com/checkstyle/checkstyle/pull/11805, due to missed tag at https://github.com/checkstyle/contribution/pull/613#discussion_r912063874

**Comment by nick-mancuso** [comment]
Generating new reports at  https://github.com/checkstyle/checkstyle/pull/11805, due to missed tag at https://github.com/checkstyle/contribution/pull/613#discussion_r912063874

File: checkstyle-tester/latest-projects-to-test-on.properties [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# List of GIT repositories to clone / pull for checking with Checkstyle
+# File format: REPO_NAME|[local|git|hg]|URL|[COMMIT_ID]|[EXCLUDE FOLDERS]
+# Please note that bash comments works in this file
+
+guava|git|https://github.com/google/guava|||
+
+spotbugs|git|https://github.com/spotbugs/spotbugs|||
```
Comment (nick-mancuso) [inline checkstyle-tester/latest-projects-to-test-on.properties:7]: Let’s use `guava-latest` naming convention for all projects here to allow for easy copy paste and no conflicts on local.

Comment (nick-mancuso) [reply checkstyle-tester/latest-projects-to-test-on.properties]: Done

**Comment by romani** [comment]
I am not sure that latest sources is stable execution, but if we want to play with it - ok.
As option it is good.

**Review by romani — APPROVED** [review]

**Review by rnveach — APPROVED** [review]
I am still not liking creating a new file for every off shot need.

**Pull request merged**